job-scheduler/src/main/java/org/hawkular/metrics/scheduler/impl/SchedulerImpl.java

```diff
@@ -361,7 +367,7 @@ public void start() {
                 queryScheduler)));
     }
 
-    private Observable<JobLock> acquireJobLock(JobDetails jobDetails) {
+    private Observable<JobLock> acquireJobLock(JobDetailsImpl jobDetails) {
```

Comment:
If I use the interface in places where I do not need any specific impl details, I wind up having to do casting which is ugly. I find it easier and cleaner to just use JobDetailsImpl everywhere rather than cast. These are all impl classes any way that are not part of the public API.